src/tensors/indexmanipulations.jl

```diff
@@ -313,23 +311,23 @@ function add_transform!(tdst::AbstractTensorMap{S,N₁,N₂},
                         β::Number,
                         backend::Backend...) where {S,N₁,N₂}
     @boundscheck begin
-        all(i -> space(tsrc, p₁[i]) == space(tdst, i), 1:N₁) ||
-            throw(SpaceMismatch("source = $(codomain(tsrc))←$(domain(tsrc)),
-            dest = $(codomain(tdst))←$(domain(tdst)), p₁ = $(p₁), p₂ = $(p₂)"))
-        all(i -> space(tsrc, p₂[i]) == space(tdst, N₁ + i), 1:N₂) ||
+        permute(space(tsrc), (p₁, p₂)) == space(tdst) ||
             throw(SpaceMismatch("source = $(codomain(tsrc))←$(domain(tsrc)),
             dest = $(codomain(tdst))←$(domain(tdst)), p₁ = $(p₁), p₂ = $(p₂)"))
     end
 
     I = sectortype(S)
     if p₁ == codomainind(tsrc) && p₂ == domainind(tsrc)
-        add!(tdst, tsrc, α, β)
+        @inbounds add!(tdst, tsrc, α, β)
     elseif I === Trivial
-        _add_trivial_kernel!(tdst, tsrc, (p₁, p₂), fusiontreetransform, α, β, backend...)
+        @inbounds _add_trivial_kernel!(tdst, tsrc, (p₁, p₂), fusiontreetransform, α, β,
```

Comment:
I don't think these `@inbounds` do anything. I use the `@inbounds` and `@boundscheck` to actually do the space checking. The `_add_...` methods don't have additional space checks, so there is not really anything to disable I think. Only `add!` has an additional space check, but that one is not indicated with `@boundscheck`, so it cannot be disabled.

Reply:
I actually did not mean to elide the space checks, but rather the checks from https://github.com/Jutho/TensorKit.jl/blob/370dd92a3a17958bd028f8211ee2464b87040511/src/tensors/tensor.jl#L604-L608
I did however definitely do this wrong, so let me try again ;)

Reply:
Ok I see. Another solution would have been to add `@propagate_inbounds` to the `_add` functions.

Reply:
I don't think it matters too much, should I still change it?

Reply:
No this is perfect.